File: backend/app/schemas/refund_request.py
from datetime import datetime
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel, Field


RefundRequestStatus = Literal["REQUESTED", "UNDER_REVIEW", "APPROVED", "REJECTED", "REFUNDED"]


class RefundRequestCreate(BaseModel):
    order_id: int
    reason: str = Field(min_length=3, max_length=500)


class RefundRequestUpdate(BaseModel):
    status: RefundRequestStatus
    internal_note: str | None = Field(default=None, max_length=500)


class RefundRequestPublic(BaseModel):
    id: int
    order_id: int
    customer_name: str
    order_total: Decimal
    status: RefundRequestStatus
    reason: str
    internal_note: str | None
    requested_by_admin_id: int
    reviewed_by_admin_id: int | None
    created_at: datetime
    reviewed_at: datetime | None
    refunded_at: datetime | None
    updated_at: datetime

File: frontend/src/types/refundRequest.ts
export type RefundRequestStatus =
  | "REQUESTED"
  | "UNDER_REVIEW"
  | "APPROVED"
  | "REJECTED"
  | "REFUNDED";

export interface RefundRequest {
  id: number;
  order_id: number;
  customer_name: string;
  order_total: number;
  status: RefundRequestStatus;
  reason: string;
  internal_note: string | null;
  requested_by_admin_id: number;
  reviewed_by_admin_id: number | null;
  created_at: string;
  reviewed_at: string | null;
  refunded_at: string | null;
  updated_at: string;
}
